src/Http/ServerRequest.php

```diff
@@ -1743,9 +1743,44 @@ public function withProtocolVersion($version)
     }
 
     /**
+     * Get a value from the request's environment data.
+     * Fallback to using env() if the key is not set in the $environment property.
+     *
+     * @param string $key The key you want to read from.
+     * @param string|null $default Default value when trying to retrieve an environment
+     *   variable's value that does not exist.
+     * @return string|null Either the environment value, or null if the value doesn't exist.
+     */
+    public function getEnv($key, $default = null)
+    {
+        $key = strtoupper($key);
+        if (!array_key_exists($key, $this->_environment)) {
+            $this->_environment[$key] = env($key);
+        }
+
+        return $this->_environment[$key] !== null ? $this->_environment[$key] : $default;
+    }
+
+    /**
+     * Set a value to the request's environment data.
+     *
+     * @param string $key The key you want to write to.
+     * @param string $value Value to set
+     * @return $this
+     */
+    public function setEnv($key, $value)
```

Comment:
Given that requests are trending towards immutable patterns, shouldn't any new methods follow that style?